hack/generator/pkg/codegen/pipeline/json_serialization_test_cases_test.go

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package pipeline
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+
+	"github.com/Azure/azure-service-operator/hack/generator/pkg/astmodel"
+	"github.com/Azure/azure-service-operator/hack/generator/pkg/test"
+)
+
+func TestInjectJsonSerializationTests(t *testing.T) {
+	g := NewGomegaWithT(t)
+
+	idFactory := astmodel.NewIdentifierFactory()
+	// Test Resource V1
+
+	specV1 := test.CreateSpec(
+		test.Pkg2020,
+		"Person",
+		test.FullNameProperty,
+		test.FamilyNameProperty,
+		test.KnownAsProperty)
+	statusV1 := test.CreateStatus(test.Pkg2020, "Person")
+	resourceV1 := test.CreateResource(test.Pkg2020, "Person", specV1, statusV1)
+
+	// Test Resource V2
+
+	specV2 := test.CreateSpec(
```

Comment:
Do we need both v1 and v2 in this test? It feels like just v1 or just v2 (which results in v1 + v1storage or v2 + v2storage) would be enough to test the generation of these tests? Probably v2 because it has more properties with different types?

I don't see anything fundamentally different between the two different "cases" (v1 and v2) here, and it seems strange to include that aspect of things but not other aspects that "generate types" such as ARM. Am I missing a special interaction that multiple versions will have with the test generation?

Reply:
This is the smallest test case I could think of that would hit all three scenarios:

* API Type
* Storage type (non-hub)
* Storage type (hub)

Reply:
I've added a comment.